MetaU Capstone Project/Models/MusicSession.m

```diff
@@ -241,5 +242,111 @@ + (void)removeFromQueue: ( NSString * )sessionCode index:( NSInteger )trackIndex
     }];
 }
 
++ (void)addToPlayedTracks: ( NSString * )sessionCode track:(NSDictionary *)trackInfo withCompletion: ( PFBooleanResultBlock _Nullable ) completion {
+    
+    PFQuery *query = [[MusicSession query] whereKey:@"sessionCode" equalTo:sessionCode];
+
+    NSDictionary *task = @{
+        @"track" : trackInfo,
+        @"addedBy" : PFUser.currentUser
+    };
+    
+    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable session, NSError * _Nullable error) {
+        if (session) {
+            NSMutableArray *track = [session[0] valueForKey:@"queue"][0];
+            NSMutableArray *playedTracks = [session[0] valueForKey:@"playedTracks"];
+
+            [playedTracks addObject:task];
+            [session setValue:playedTracks forKey:@"playedTracks"];
+            
+            if ([[track valueForKey:@"URI"] isEqual:[trackInfo valueForKey:@"URI"]]) {
+                [MusicSession removeFromQueue:sessionCode index:0 withCompletion:nil];
+            }
+            
+            [PFObject saveAllInBackground:session];
+        } else {
+            NSLog(@"Error getting session: %@", error.localizedDescription);
+        }
+    }];
+}
+
++ (void)updateCurrentlyPlaying: ( NSString * )sessionCode track:( NSDictionary * )trackInfo withCompletion: ( PFBooleanResultBlock _Nullable ) completion {
+ 
+    PFQuery *query = [[MusicSession query] whereKey:@"sessionCode" equalTo:sessionCode];
+    PFUser *user = [PFUser currentUser];
+
+    NSDictionary *task = @{
+        @"track" : trackInfo,
+        @"addedBy" : user
+    };
+
+    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable session, NSError * _Nullable error) {
+        if (session) {
+            NSMutableArray *queue = [session[0] valueForKey:@"queue"];
```

Comment:
Not to sound like a broken record, but you should consider making allllll of these strings constant.
- "queue"
- "track"
- "addedBy"
- etc...